src/common/enums/join_type.cpp

```diff
@@ -8,7 +8,8 @@ bool IsLeftOuterJoin(JoinType type) {
 }
 
 bool IsRightOuterJoin(JoinType type) {
```

Comment:
Similarly this feels wrong - `RIGHT_SEMI` is not a right outer join, and this now leads to e.g. the statistics propagation stating that `NULL` values can be introduced in the RHS which is not the case for `RIGHT_SEMI/RIGHT_ANTI`.